darwin: curlie 1.6.9 no longer builds

### Steps To Reproduce
Steps to reproduce the behavior:
1. nix build -L 'nixpkgs#curlie'

### Build log

<details>

```
building '/nix/store/7q1srqrkag4jpdszpv7kbq0c19dki0dd-curlie-1.6.9.drv'...
curlie> unpacking sources
curlie> unpacking source archive /nix/store/ipkbl8gfpqyx71xjh2i1awyhbmfrf00c-source
curlie> source root is source
curlie> patching sources
curlie> updateAutotoolsGnuConfigScriptsPhase
curlie> configuring
curlie> building
curlie> Building subPackage .
curlie> internal/unsafeheader
curlie> internal/goos
curlie> internal/race
curlie> internal/goarch
curlie> internal/goexperiment
curlie> unicode
curlie> sync/atomic
curlie> internal/cpu
curlie> internal/itoa
curlie> unicode/utf8
curlie> math/bits
curlie> internal/abi
curlie> runtime/internal/math
curlie> runtime/internal/sys
curlie> encoding
curlie> unicode/utf16
curlie> math
curlie> runtime/internal/atomic
curlie> internal/bytealg
curlie> runtime
curlie> sync
curlie> internal/reflectlite
curlie> internal/testlog
curlie> errors
curlie> sort
curlie> internal/oserror
curlie> path
curlie> io
curlie> strconv
curlie> syscall
curlie> bytes
curlie> strings
curlie> reflect
curlie> regexp/syntax
curlie> internal/syscall/execenv
curlie> internal/syscall/unix
curlie> time
curlie> regexp
curlie> github.com/rs/curlie/formatter
curlie> io/fs
curlie> internal/poll
curlie> context
curlie> golang.org/x/sys/unix
curlie> # golang.org/x/sys/unix
curlie> vendor/golang.org/x/sys/unix/syscall_darwin.1_13.go:25:3: //go:linkname must refer to declared function or variable
curlie> vendor/golang.org/x/sys/unix/zsyscall_darwin_arm64.1_13.go:27:3: //go:linkname must refer to declared function or variable
curlie> vendor/golang.org/x/sys/unix/zsyscall_darwin_arm64.1_13.go:40:3: //go:linkname must refer to declared function or variable
curlie> vendor/golang.org/x/sys/unix/zsyscall_darwin_arm64.go:28:3: //go:linkname must refer to declared function or variable
curlie> vendor/golang.org/x/sys/unix/zsyscall_darwin_arm64.go:43:3: //go:linkname must refer to declared function or variable
curlie> vendor/golang.org/x/sys/unix/zsyscall_darwin_arm64.go:59:3: //go:linkname must refer to declared function or variable
curlie> vendor/golang.org/x/sys/unix/zsyscall_darwin_arm64.go:75:3: //go:linkname must refer to declared function or variable
curlie> vendor/golang.org/x/sys/unix/zsyscall_darwin_arm64.go:90:3: //go:linkname must refer to declared function or variable
curlie> vendor/golang.org/x/sys/unix/zsyscall_darwin_arm64.go:105:3: //go:linkname must refer to declared function or variable
curlie> vendor/golang.org/x/sys/unix/zsyscall_darwin_arm64.go:121:3: //go:linkname must refer to declared function or variable
curlie> vendor/golang.org/x/sys/unix/zsyscall_darwin_arm64.go:121:3: too many errors
curlie> internal/fmtsort
curlie> os
curlie> encoding/binary
curlie> encoding/base64
curlie> io/ioutil
curlie> path/filepath
curlie> fmt
curlie> os/exec
curlie> net/url
curlie> encoding/json
curlie> github.com/rs/curlie/args
builder for '/nix/store/7q1srqrkag4jpdszpv7kbq0c19dki0dd-curlie-1.6.9.drv' failed with exit code 1; last 10 log lines:
  os
  encoding/binary
  encoding/base64
  io/ioutil
  path/filepath
  fmt
  os/exec
  net/url
  encoding/json
  github.com/rs/curlie/args
error: build of '/nix/store/7q1srqrkag4jpdszpv7kbq0c19dki0dd-curlie-1.6.9.drv' failed
```

</details>

### Additional context

Not a go developer. I think the issue is the same as #182108. That requires a fix from upstream; I don't know whether this is the same, though, and my attempts to patch the build (and thus demonstrate it needs an upstream fix) have been unsuccessful.

I used `overrideAttrs` with the following patch.

```diff
diff --git 1/go.mod 2/go.mod
index 660463a..5249458 100644
--- 1/go.mod
+++ 2/go.mod
@@ -2,7 +2,7 @@ module github.com/rs/curlie

 require (
   golang.org/x/crypto v0.0.0-20180524125353-159ae71589f3
-  golang.org/x/sys v0.0.0-20191026070338-33540a1f6037
+  golang.org/x/sys v0.0.0-20220913175220-63ea55921009
 )

 go 1.13
```

This resulted in the following error.

```console
  go: inconsistent vendoring in /private/tmp/nix-build-curlie-1.6.9.drv-0/source:
       golang.org/x/sys@v0.0.0-20220913175220-63ea55921009: is explicitly required in go.mod, but vendor/modules.txt indicates golang.org/x/sys@v0.0.0-20191026070338-33540a1f6037

         To ignore the vendor directory, use -mod=readonly or -mod=mod.
         To sync the vendor directory, run:
             go mod vendor
```

I tried vendoring the derivation and adding the call to `go mod vendor` (using `overrideModAttrs`) but at this point I hit the time limit I'd set to investigate this.

### Notify maintainers

@Ma27 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"aarch64-darwin"`
 - host os: `Darwin 21.6.0, macOS 12.5.1`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.9.1`
 - channels(root): `"nixpkgs-21.11pre300649.3e0ce8c5d47"`
 - nixpkgs: `/nix/store/bvrq57jdn9wjmjz765v5wpkp3n3j236b-nixpkgs`
```

--- a/pkgs/tools/networking/curlie/bump-golang-x-sys.patch
+++ b/pkgs/tools/networking/curlie/bump-golang-x-sys.patch
@@ -1,0 +1,13 @@
+diff --git a/go.mod b/go.mod
+index 660463a..5249458 100644
+--- a/go.mod
++++ b/go.mod
+@@ -2,7 +2,7 @@ module github.com/rs/curlie
+ 
+ require (
+ 	golang.org/x/crypto v0.0.0-20180524125353-159ae71589f3
+-	golang.org/x/sys v0.0.0-20191026070338-33540a1f6037
++	golang.org/x/sys v0.0.0-20211110154304-99a53858aa08
+ )
+ 
+ go 1.13

--- a/pkgs/tools/networking/curlie/default.nix
+++ b/pkgs/tools/networking/curlie/default.nix
@@ -11,7 +11,11 @@ buildGoModule rec {
     hash = "sha256-3EKxuEpFm+lp2myMfymYYY9boSXGOF2iAdjtGKnjJK0=";
   };
 
-  vendorSha256 = "sha256-tYZtnD7RUurhl8yccXlTIvOxybBJITM+it1ollYJ1OI=";
+  patches = [
+    ./bump-golang-x-sys.patch
+  ];
+
+  vendorSha256 = "sha256-VsPdMUfS4UVem6uJgFISfFHQEKtIumDQktHQFPC1muc=";
 
   ldflags = [ "-s" "-w" "-X main.version=${version}" ];
 

